src/d3-ext/profile.js

```diff
@@ -0,0 +1,316 @@
+/**
+ * @fileoverview Provides a D3js component to be used to draw an elevation
+ * profile chart.
+ *
+ * Example usage:
+ *
+ * var selection = d3.select('#element_id');
+ * var profile = ngeo.profile().onHover(function(point) {
+ *   console.log(point.x, point.y);
+ * }).onOut(function() {console.log("out");});
+ * selection.datum(data).call(profile);
+ *
+ *
+ * The selection data to be used should look like the following:
+ *
+ * [
+ *     {
+ *         "y": 199340,
+ *         "values": {"mnt": 788.7},
+ *         "dist": 0.0,
+ *         "x": 541620
+ *     }, ...
+ * ]
+ *
+ */
+goog.provide('ngeo.profile');
+
+
+/**
+ *
+ * @return {Object}
+ * @export
+ *
+ */
+ngeo.profile = function() {
```

Comment:
Indeed. Do you want me to give it a try? Or can you fork my branch and work on this yourself?